packages/@sanity/default-layout/src/navbar/search/components/SearchResults.tsx

```diff
@@ -0,0 +1,130 @@
+// @todo: remove the following line when part imports has been removed from this file
+///<reference types="@sanity/types/parts" />
+
+import {Box} from '@sanity/ui'
+import {getPublishedId} from 'part:@sanity/base/util/draft-utils'
+import React, {Dispatch, SetStateAction, useCallback, useEffect, useRef} from 'react'
+import {useVirtual} from 'react-virtual'
+import styled from 'styled-components'
+import {VIRTUAL_LIST_ITEM_HEIGHT, VIRTUAL_LIST_OVERSCAN} from '../constants'
+import {useCommandList} from '../contexts/commandList'
+import {useSearchState} from '../contexts/search'
+import {NoResults} from './NoResults'
+import {PointerOverlay} from './PointerOverlay'
+import {SearchError} from './SearchError'
+import {SearchResultItem} from './searchResultItem'
+
+interface SearchResultsProps {
+  onClose: () => void
+  setChildContainerRef: Dispatch<SetStateAction<HTMLDivElement>>
+  setPointerOverlayRef: Dispatch<SetStateAction<HTMLDivElement>>
+}
+
+export function SearchResults({
+  onClose,
+  setChildContainerRef,
+  setPointerOverlayRef,
+}: SearchResultsProps) {
+  const {
+    dispatch,
+    recentSearchesStore,
+    state: {terms, result},
+  } = useSearchState()
+
+  const childParentRef = useRef()
+
+  const {scrollToIndex, totalSize, virtualItems} = useVirtual({
+    estimateSize: useCallback(() => VIRTUAL_LIST_ITEM_HEIGHT, []),
+    overscan: VIRTUAL_LIST_OVERSCAN,
+    parentRef: childParentRef,
+    size: result.hits.length,
+  })
+
+  const {
+    onChildClick,
+    onChildMouseDown,
+    onChildMouseEnter,
+    setVirtualListScrollToIndex,
+  } = useCommandList()
+
+  /**
+   * Send react-virtual's `scrollToIndex` function to shared CommandList context
+   */
+  useEffect(() => {
+    setVirtualListScrollToIndex(scrollToIndex)
+    return () => setVirtualListScrollToIndex(null)
+  }, [setVirtualListScrollToIndex, scrollToIndex])
+
+  /**
+   * Add current search terms to recent searches, trigger child item click and close search
+   */
+  const handleResultClick = useCallback(() => {
+    // Add terms to Local Storage
+    const updatedRecentSearches = recentSearchesStore?.addSearchTerm(terms)
+    dispatch({
+      recentSearches: updatedRecentSearches,
+      type: 'RECENT_SEARCHES_SET',
+    })
+    onChildClick?.()
+    onClose()
+  }, [dispatch, onChildClick, onClose, recentSearchesStore, terms])
+
+  return (
+    <SearchResultsWrapper aria-busy={result.loading} $loading={result.loading}>
+      <PointerOverlay ref={setPointerOverlayRef} />
+
+      {result.error ? (
+        <SearchError />
+      ) : (
+        <>
+          {!!result.hits.length && (
+            // (Has search results)
+            <VirtualList ref={childParentRef}>
```

Comment:
I think `VirtualList as="ul"` and `SearchResultItemWrapper as="li"` would be more semantic markup here :)